From 6fb8e89af2adb97833c4c04df4a40d433cd506e3 Mon Sep 17 00:00:00 2001
From: Eugeniy Paltsev <Eugeniy.Paltsev@synopsys.com>
Date: Mon, 23 Sep 2019 12:53:32 +0300
Subject: [PATCH 3/3] ARC: HSDK: enable tunnel

Enables tunnel from 0x1000_0000-0x7FFF_FFFF on HSDK physical to map
to 0x1000_0000-0x7FFF_FFFF on HAPS physical.

Signed-off-by: Eugeniy Paltsev <Eugeniy.Paltsev@synopsys.com>
---
 arch/arc/plat-hsdk/platform.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/arch/arc/plat-hsdk/platform.c b/arch/arc/plat-hsdk/platform.c
index 0b961a2a10b8..bf86bcb1322a 100644
--- a/arch/arc/plat-hsdk/platform.c
+++ b/arch/arc/plat-hsdk/platform.c
@@ -246,9 +246,10 @@ static void __init hsdk_init_memory_bridge(void)
 	 */
 	reg = readl(CREG_AXI_M_HS_CORE_BOOT) & (~0x3);
 	writel(reg, CREG_AXI_M_HS_CORE_BOOT);
-	writel(0x11111111, CREG_AXI_M_SLV0(M_HS_CORE));
+	/* Tunnel 0x1000_0000-0x7FFF_FFFF*/
+	writel(0x33333331, CREG_AXI_M_SLV0(M_HS_CORE));
 	writel(0x63111111, CREG_AXI_M_SLV1(M_HS_CORE));
-	writel(0xFEDCBA98, CREG_AXI_M_OFT0(M_HS_CORE));
+	writel(0x76543218, CREG_AXI_M_OFT0(M_HS_CORE));
 	writel(0x0E543210, CREG_AXI_M_OFT1(M_HS_CORE));
 	writel(UPDATE_VAL, CREG_AXI_M_UPDT(M_HS_CORE));
 
-- 
2.21.0

